pkg/auth/monitor/monitor.go

```diff
@@ -47,9 +59,61 @@ func (dm *dropMonitor) NotifyPerfEvent(data []byte, cpu int) {
 	// Packet data starts right after the DropNotify
 	connInfo := monitor.GetConnectionInfo(data[unsafe.Sizeof(*dn):])
```

Comment:
Not your code, but consider moving this logic (including the `binary.Read`) to the other side of the queue. Both the decoding and `monitor.GetConnectionInfo` are relatively expensive operations.

Also, tiny nit (again, not your code, I'm sorry): Instead of `unsafe.Sizeof(*dn)`, consider just using ` monitor.DropNotifyLen` (which is what we use in Hubble).

Reply:
thanks for the hints. moved it to the other side - with the byte slice becoming the only part of the event now.